jobs/furnishings/src/furnishings/stage_processors/stage_three.py

```diff
@@ -32,15 +40,15 @@ def process(app: Flask):
         )
         batch_processings = (
             db.session.query(BatchProcessing)
-            .filter(BatchProcessing.status == BatchProcessing.BatchProcessingStatus.PROCESSING)
+            .filter(BatchProcessing.status == BatchProcessing.BatchProcessingStatus.COMPLETED)
             .filter(BatchProcessing.step == BatchProcessing.BatchProcessingStep.DISSOLUTION)
             .filter(Batch.id == BatchProcessing.batch_id)
             .filter(Batch.batch_type == Batch.BatchType.INVOLUNTARY_DISSOLUTION)
-            .filter(Batch.status == Batch.BatchStatus.PROCESSING)
-            .filter(not_(furnishing_subquery))
+            .filter(has_stage_2_furnishing)
+            .filter(not_(has_stage_3_furnishing))
```

Comment:
Once stage 3 of the involuntary dissolutions job is run, the `batch_processing.status` gets [changed to `QUEUED`](https://github.com/leodube-aot/lear/blob/main/jobs/involuntary-dissolutions/involuntary_dissolutions.py#L284), and then shortly after changed to `COMPLETED` by the filer. This means I don't think we can use `BatchProcessing.status == PROCESSING` to get the batch_processings for the furnishing run.

Here I suggest checking if the batch_processing entry _does have_ a stage 2 furnishing entry, and _does not have_ a stage 3 furnishing entry.